R/module_filter_data.R

```diff
@@ -52,7 +52,15 @@ srv_filter_data <- function(id, datasets, active_datanames, data_rv, is_active)
 
 #' @rdname module_filter_data
 .make_filtered_teal_data <- function(modules, data, datasets = NULL, datanames) {
-  data <- eval_code(data, sprintf("%1$s._raw_ <- %1$s", datanames))
+  data <- eval_code(
+    data,
+    paste0(
+      ".raw_data <- list2env(list(",
+      toString(sprintf("%1$s = %1$s", datanames)),
+      "))\n",
+      "lockEnvironment(.raw_data) #@linksto .raw_data" # this is environment and it is shared by qenvs. CAN'T MODIFY!
+    )
+  )
```

Comment:
> can we work on expressions please instead of strings - that does not look pro

We can't because `#@linksto` comment will be discarded when turned into expression. 
Also creating an expression `.raw_data <- list2env(list(<datanames> = <datanames>))` where datanames has `length > 1` will probably involve some `str2lang` or `parse(text = )`

> can we make a single expr lockEnvir(list2env(...)) - just to limit the technical clutter in SRC from two lines into a single one

we need to assign `.raw_data` in this env. If you wish it, I can make `lockEnvironment(.raw_data <- ...)`. But this is  even clutter-ous.

P.S. What I think instead we can give up keeping this code at all. But this is a danger zone as we provide object which is not "documented" by the code. Theoretically, one can use object `.raw_data` in the module to build output. I was thinking that maybe `get_var.qenv` and `[[.qenv` shouldn't allow extracting hidden objects prefixed by `.`. Then the whole concept of not documenting `.<hidden object name>` would make sense. But this would involve also changing `validate(<qenv>)` function, which checks `@code => @env`

Reply:
I'm sorry for my naive question - why do we care about `#@linksto` comment?

> where datanames has length > 1 will probably involve some str2lang or parse(text = )

This is exactly the use case where rlang shines but I do understand that you want to stick to the base. ok let's leave it

> But this is even clutter-ous.

Agree - let's leave it.

> P.S.

I see your point here but on the other hand I do like how simple qenv is. For me `get_var.qenv` is the same as `base::get()`. That _hidding_ dot vars is happening only in `ls()` but not in `get()`.
Not to mention changes in validate.
Let's keep it simple and eliminate workarounds.
If we are thinking that far - I would be going in the direction of storing these raw data outside of qenv. AFAIK these are needed for filtering so it would make sense to keep it in the dataset-filter instance.
Also, I don't think we need _complete_ datasets with raw values. We might only need colnames, distribution data, etc. But this is a complitely separate topic.

Reply:
> I'm sorry for my naive question - why do we care about #@linksto comment?

So that this is line 

```
lockEnvironment(.raw_data)
````

is extracted when you ask for the [`teal.data::get_code(data, datanames =`](https://github.com/insightsengineering/teal/pull/1342/files#diff-4f959897d64b51a6ac57c9016430bdd92163a17fa62b5a61707a4b3bc7dd6325R59) for raw objects.

When you specify datanames, then all the code needed to reproduce this object is returned. If it's hard to determine lines that have an affect on the object, because this is the case of a line that only make a side-effect on the object, then you need to mark this line with `#@linksto` tag. Otherwise side-effects made on those objects will not be returned with the code.

Reply:
Thanks. It's all connected to my very old issue [here](https://github.com/insightsengineering/teal.code/issues/181). which reflect my thinking that the code is a code but it's a string instead. I look into the [docs](https://insightsengineering.github.io/teal.data/latest-tag/reference/get_code.html#extracting-dataset-specific-code) and it's a functionality built on the top of it. 

Honestly, I'm getting more and more concerned as there are more and more functionalities built on the top of that questionable implementation but this is definitely going beyond that PR. All good here.